### ISSUE TITLE
Add HTML Anchors (ID Attribute) to Attributes in Block Parser

### ISSUE THREAD

**Issue by crowinck** [WordPress/gutenberg#28757]
Currently WP_Block_Parser and parse_blocks lists block classes/block styles among a block's attributes (attr), but the id attribute (HTML Anchor) can only be retrieved by parsing the HTML in innerHTML or innerContent, even when it is set by the user in a separate field under the Advanced block settings.

Given that there are more enhancements under discussion for the HTML Anchor (see #26685 ), and it will be important to any TOC features, it seems like the anchor should be included in a blocks listed attributes during parsing for easier php & server-side retrieval.

**Comment by talldan** [comment]
@crowinck It'd be great if you use the provided issue templates when reporting issues.

**Comment by jmesserve** [comment]
Any update on this?

**Comment by Soean** [comment]
@jmesserve No, not yet. You can work on a solution of this issue, if you want.

**Comment by talldan** [comment]
Testing a snippet like this today:
```js
wp.blocks.parse( `<!-- wp:paragraph {"className":"test"} -->
<p class="test" id="testalso">test <a href="http://localhost:8888/?page_id=2" data-type="page" data-id="2">sometest</a> here</p>
<!-- /wp:paragraph -->` )
```

and it does return a value for `attributes.anchor`, so I this can be considered solved.

The anchor comes from this attribute declaration here:
https://github.com/WordPress/gutenberg/blob/f103e2a81ae506d0bac00faed505a1d755033a58/packages/block-editor/src/hooks/anchor.js#L23-L28

This is injected into blocks that have `anchor` support.

I think this is the right way to handle it. Block should declare via attributes what html attributes/values should be parsed and stored in block attributes.
